crates/driver/src/infra/simulator/mod.rs

```diff
@@ -189,14 +191,14 @@ where
                     None
                 }
             }
-            SimulatorError::Blockchain(blockchain::Error::Gas(_)) => None,
-            SimulatorError::Blockchain(blockchain::Error::Response(_)) => None,
+            SimulatorError::Blockchain(blockchain::Error::Gas(_)) => Some(tx),
+            SimulatorError::Blockchain(blockchain::Error::Response(_)) => Some(tx),
```

Comment:
Why these three are now classified as revert errors?

Reply:
Yeah, these seem more like they should be `Other` errors. Meaning it's entirely possible that the simulated transaction is fine but we just encountered an internal error.

Reply:
You are right about Gas (I thought it was gas amount estimation related, but it's purely gas price related which the transaction has no influence on). 
However for `blockchain::Error::Response` it only happens when we fail to estimate the `accessList` which happens if the transaction doesn't simulate properly.

Will rename these variants to make that more clear.